### ISSUE TITLE
ListView ItemClick handling

### ISSUE THREAD

**Issue by takacsalbert** [CalciumFramework/Calcium#14]
Hi,

As mentioned in [https://github.com/CodonFramework/Codon/issues/13](url), you have to have a handler in your ViewModel with the following definition:
`public void HandleItemClick(object myViewModel, Android.Widget.AdapterView.ItemClickEventArgs e)`

Since my viewmodels are in a portable library, I don't want to have a reference to any Android types. Is it possible to have a simple parameter (or parameters) in the definition? I think the only useful property would be the position. But ideal solution is having the selected object passed. ;)

**Comment by DVaughan** [comment]
Not at the moment. Though I'd also like to have a more elegant solution. 
You can add a third parameter (of type View) to the handler and the view will be passed to the handler; though I realize that's not what you're after.
The trouble is that the data-binding is 'external' to the ListView and the ListView item's view doesn't necessarily retain a reference to a view-model. For now I'd recommend subscribing to the ListView ItemClick event and notifying the view-model from the view/activity.

**Comment by takacsalbert** [comment]
I understand the problem.
I said earlier that my ViewModels should be kept in a portable library for greater code reuse and I want them to be platform-independent.
So for now, I made a workaround, which is the following:
I made a wrapper that derives from the ViewModel. I wrote the ItemClick handler like mentioned with Android.Widget.AdapterView.ItemClickEventArgs. Then I get the selected object and pass it to the selected method to the original ViewModel. I replaced the the ViewModel in the IOC to the wrapper and everything works as expected without modifying the original ViewModel.
I know it is not that nice and I am thinking about a different solution. Extension methods would be much nicer, but that doesn't work with the binding. "Method Not Found" exception is thrown when I try to bind to an extension method.
I hope it makes sense and would be helpful to others. Probably you would consider adding support for extension methods to Codon. ([](https://stackoverflow.com/questions/299515/reflection-to-identify-extension-methods) or something like that)
